### ISSUE TITLE
RadioButton accessibility

### ISSUE THREAD

**Issue by qroll** [LifeSG/react-design-system#814]
**Component**
`RadioButton`

**Describe the feature request**

- Align the RadioButton with the accessibility behaviour defined in the Figma specs
- Update the Storybook example for "Multiple Options" to follow the example in Figma and include the suggested focus and reading order

**Acceptance criteria**

- Selected and disabled state should be correctly read out by the AT
- `radio` role should be correctly read out by the AT
- Keyboard support is included
- No extraneous elements are read out
- In the Storybook example, labels should be correctly associated with the radio button
- Minimally test in NVDA + Chrome (Windows), VO + Safari (macOS), Talkback + Chrome (Android)

**References**
[Figma](https://www.figma.com/design/wDAxbRmlq4xOn4UJ8zPuEF/%F0%9F%A7%91%F0%9F%8F%BB%E2%80%8D%F0%9F%A6%AF-Flagship-A11y-Annotations?node-id=159-468&p=f&m=dev)
[Radio Storybook](https://designsystem.life.gov.sg/react/index.html?path=/docs/selection-and-input-radiobutton--docs)
